tmail-backend/event-bus-redis/src/test/java/org/apache/james/events/RedisEventBusServiceSentinelTest.java

```diff
@@ -0,0 +1,117 @@
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package org.apache.james.events;
+
+import static org.apache.james.events.EventBusTestFixture.EVENT;
+import static org.apache.james.events.EventBusTestFixture.KEY_1;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.james.backends.rabbitmq.RabbitMQExtension;
+import org.apache.james.backends.redis.RedisClientFactory;
+import org.apache.james.backends.redis.RedisSentinelExtension;
+import org.apache.james.backends.redis.SentinelRedisConfiguration;
+import org.apache.james.metrics.tests.RecordingMetricFactory;
+import org.apache.james.server.core.filesystem.FileSystemImpl;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import reactor.core.publisher.Mono;
+
+public class RedisEventBusServiceSentinelTest {
+    @RegisterExtension
+    static RabbitMQExtension rabbitMQExtension = RabbitMQExtension.singletonRabbitMQ()
+        .isolationPolicy(RabbitMQExtension.IsolationPolicy.WEAK);
+
+    @RegisterExtension
+    static RedisSentinelExtension redisExtension = new RedisSentinelExtension();
+
+    private RabbitMQAndRedisEventBus eventBus;
+
+    @AfterEach
+    void tearDown() {
+        redisExtension.getRedisSentinelCluster().redisMasterReplicaContainerList().unPauseMasterNode();
+
+        try {
+            eventBus.stop();
+        } catch (Exception e) {
+            // Ignore exception
+        }
+    }
+
+    @Test
+    void dispatchShouldNotThrowWhenRedisMasterDownAndSentinelFailoverProcessKicksIn() throws Exception {
+        // Given EventBus with failureIgnore set to false
+        RedisEventBusConfiguration redisEventBusConfiguration = new RedisEventBusConfiguration(false, Duration.ofSeconds(3));
+        initEventBus(redisEventBusConfiguration);
+
+        EventCollector listener = new EventCollector();
+        EventBusTestFixture.GroupA registeredGroup = new EventBusTestFixture.GroupA();
+        eventBus.register(listener, registeredGroup);
+
+        //When Redis down
+        redisExtension.getRedisSentinelCluster().redisMasterReplicaContainerList().pauseMasterNode();
+        Thread.sleep(500);
```

Comment:
If I add this 
```
        Thread.sleep(10000L); // enough time to trigger the failover process

        redisExtension.getRedisSentinelCluster().redisMasterReplicaContainerList().unPauseMasterNode(); // bring back the old master node (now should be replica)
```
And use the old code previous the fix, then these two failover tests still pass.

That makes me surprised. 

Open question: Any chance the failover has been work with sentinel mode before, it is just we configured the wrong topology cf https://github.com/linagora/tmail-backend/issues/1585#issuecomment-2775174463?

Reply:
Hmmmm... are you sure it's enough time? It looked like locally the test would take around 20s total to run for example so my guess is failover maybe a bit longer? I could take a look, but i would test this more with 20 or 30s to be sure. You might have a master node coming back online here before the sentinel node decides of a new master. 

But yeah let's try

Reply:
Same result as you when I wait 30s.

I would guess likely due to standalone connection like you yes.